src/lib/storage/dictionary_segment/dictionary_segment_iterable.hpp

```diff
@@ -10,16 +10,17 @@
 
 namespace hyrise {
 
-template <typename T, typename Dictionary>
-class DictionarySegmentIterable : public PointAccessibleSegmentIterable<DictionarySegmentIterable<T, Dictionary>> {
+template <typename T, typename Dictionary, typename DictionarySpan>
+class DictionarySegmentIterable
+    : public PointAccessibleSegmentIterable<DictionarySegmentIterable<T, Dictionary, DictionarySpan>> {
```

Comment:
Will the `_dictionary` member be removed in the future or is the `_dictionary_span` lacking required features?

Reply:
Of course.
But first, the memory of all segments need to be managed by the storage manager.